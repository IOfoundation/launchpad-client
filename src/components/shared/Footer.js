import React from 'react';
import {Link} from 'react-router';
import Logo from './Logo';
import FacebookIcon from '../shared/FacebookIcon';
import TwitterIcon from '../shared/TwitterIcon';
import LinkedinIcon from '../shared/LinkedinIcon';
import GitIcon from '../shared/GitIcon';

const Footer = () => {
  return (
    <footer className="main-footer text-thin">
      <div className="contentContainer row">
        <div className="col-md-2 col-xs-2 text-xs-margin">
          <a className="logo-link" href="/">
            <Logo />
          </a>
        </div>
        <div
          data-test="footer-login"
          className="col-md-2 col-md-offset-1 col-xs-2 text-xs-margin"
        >
          <p className="main-footer_subtitle text-regular">{'LOGIN'}</p>
          <Link to="">{'Admin Login'}</Link>
        </div>
        <div
          data-test="footer-resources"
          className="col-md-2 col-xs-2 text-xs-margin"
        >
          <p className="main-footer_subtitle text-regular">{'RESOURCES'}</p>
          <Link to="">{'Search'}</Link>
          <Link to="">{'Events'}</Link>
          <Link to="">{'Blog'}</Link>
        </div>
        <div
          data-test="footer-about"
          className="col-md-2 col-xs-2 text-xs-margin"
        >
          <p className="main-footer_subtitle text-regular">{'ABOUT'}</p>
          <Link to="">{'About'}</Link>
          <Link to="/contact-us">{'Contact'}</Link>
          <Link to="/terms-of-use">{'Terms & Conditions'}</Link>
          <Link to="/privacy-policy">{'Privacy Policy'}</Link>
        </div>
        <div className="social-icons col-md-2 col-xs-2 text-xs-margin">
          <p className="main-footer_subtitle text-regular">{'SOCIAL'}</p>
          <Link className="main-footer_icon" to="">
            <TwitterIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <FacebookIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <LinkedinIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <GitIcon />
          </Link>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
