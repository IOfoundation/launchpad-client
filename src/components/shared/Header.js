import React, {Fragment} from 'react';
import {PropTypes} from 'prop-types';
import {Link, IndexLink} from 'react-router';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import BarsIcon from '@Shared/barsIcon';

import * as blogsActions from '@Actions/blogs';
import * as userActions from '@Actions/user';
import {getAuthorization} from '@Utils';

const activeStyles = {opacity: '1'};

const Header = props => {
  let button;

  const setDefaultCategory = () => {
    props.actions.setCategory('front page');
  };

  const signOut = event => {
    const {user, Authorization, router} = props;

    event.preventDefault();
    user.signOut(Authorization, true);
    router.push('/admin-login');
  };

  if (props.isAuth) {
    button = (
      <Fragment>
        <a className="header_link white-link" onClick={signOut}>
          {'Logout'}
        </a>
        <Link className="header_link white-link" to="/admin/profile">
          {'Admin'}
        </Link>
      </Fragment>
    );
  } else {
    button = (
      <Link
        activeStyle={activeStyles}
        className="header_link white-link"
        to="/admin-login"
      >
        {'Admin Login'}
      </Link>
    );
  }

  return (
    <header className="headerContainer" htmlFor="header-dropdown">
      <div className="contentContainer header grid between-xs middle-xs text-semi">
        <div className="header_links-title">
          <span className="header_title">
            <IndexLink
              to="/"
              className="white-link text-bold"
              style={activeStyles}
            >
              {'Welcome to Resource Finder'}
            </IndexLink>
          </span>
        </div>
        <input id="header-dropdown" type="checkbox" name="dropdown" />
        <label htmlFor="header-dropdown" className="collapse-menu-icon">
          <BarsIcon size={24} />
        </label>
        <nav className="header_links-contact text-bold">
          <Link
            activeStyle={activeStyles}
            className="header_link white-link "
            to="/businesses"
          >
            {'Resources'}
          </Link>
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/events"
          >
            {'Events'}
          </Link>
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/blog"
            onClick={setDefaultCategory}
          >
            {'Blog'}
          </Link>
          {button}
        </nav>
      </div>
    </header>
  );
};

const mapStateToProps = _state => {
  const Authorization = getAuthorization(_state);

  return {
    isAuth: Authorization !== '',
    Authorization,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    actions: bindActionCreators(blogsActions, _dispatch),
    user: bindActionCreators(userActions, _dispatch),
  };
};

Header.propTypes = {
  actions: PropTypes.shape({
    setCategory: PropTypes.func,
  }),
  Authorization: PropTypes.string,
  homePage: PropTypes.bool.isRequired,
  isAuth: PropTypes.bool,
  router: PropTypes.shape({
    push: PropTypes.func,
  }),
  user: PropTypes.shape({
    signOut: PropTypes.func,
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
