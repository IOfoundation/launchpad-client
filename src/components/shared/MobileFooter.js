import React from 'react';
import {Link} from 'react-router';
import ArrowRight from '../shared/ArrowRight';
import FacebookIcon from '../shared/FacebookIcon';
import TwitterIcon from '../shared/TwitterIcon';
import LinkedinIcon from '../shared/LinkedinIcon';
import GitIcon from '../shared/GitIcon';

const MobileFooter = () => {
  return (
    <footer className="mobile-footer row grid">
      <ul className="mobile-footer-list">
        <input id="dropDown-1" type="checkbox" name="dropdown" />
        <li className="list-options">
          <label htmlFor="dropDown-1">{'LOGIN'}</label>
          <ArrowRight
            htmlFor="dropDown-1"
            className="right-arrow"
            size={20}
            style={{color: '#fff', verticalAlign: 'middle'}}
          />
          <ul className="subDropDown">
            <Link to="/admin-login">{'Admin Login'}</Link>
          </ul>
        </li>
      </ul>
      <ul className="mobile-footer-list">
        <input id="dropDown-2" type="checkbox" name="dropdown" />
        <li className="list-options">
          <label htmlFor="dropDown-2">{'RESOURCES'}</label>
          <ArrowRight
            htmlFor="dropDown-2"
            className="right-arrow"
            size={20}
            style={{color: '#fff', verticalAlign: 'middle'}}
          />
          <ul className="subDropDown">
            <Link to="/businesses">{'Search'}</Link>
            <Link to="/events">{'Events'}</Link>
            <Link to="/blog">{'Blog'}</Link>
          </ul>
        </li>
      </ul>
      <ul className="mobile-footer-list">
        <input id="dropDown-3" type="checkbox" name="dropdown" />
        <li className="list-options">
          <label htmlFor="dropDown-3">{'ABOUT'}</label>
          <ArrowRight
            htmlFor="dropDown-3"
            className="right-arrow"
            size={20}
            style={{color: '#fff', verticalAlign: 'middle'}}
          />
          <ul className="subDropDown">
            <Link to="">{'About'}</Link>
            <Link to="/contact-us">{'Contact'}</Link>
            <Link to="/terms-of-use">{'Terms & Conditions'}</Link>
            <Link to="/privacy-policy">{'Privacy Policy'}</Link>
          </ul>
        </li>
      </ul>
      <ul className="mobile-footer-list social-icons-container">
        <li className="social-icons-title">{'SOCIAL'}</li>
        <li>
          <Link className="main-footer_icon" to="">
            <TwitterIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <FacebookIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <LinkedinIcon />
          </Link>
          <Link className="main-footer_icon" to="">
            <GitIcon />
          </Link>
        </li>
      </ul>
    </footer>
  );
};

export default MobileFooter;
