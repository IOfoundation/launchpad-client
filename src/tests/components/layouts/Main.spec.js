import React from 'react';
import {shallow} from 'enzyme';

import MainLayout from '../../../components/layouts/Main';
import Header from '../../../components/shared/Header';
import Footer from '../../../components/shared/Footer';
import MobileFooter from '../../../components/shared/MobileFooter';

describe('<MainLayout />', () => {
  it('Render the Header component', () => {
    const wrapper = shallow(<MainLayout />);
    expect(wrapper.find(Header).length).toEqual(1);
  });

  it('Render the Footer component', () => {
    const wrapper = shallow(<MainLayout />);
    expect(wrapper.find(Footer).length).toEqual(1);
  });

  it('Render the childrens added to the layout', () => {
    const wrapper = shallow(
      <MainLayout>
        <div className={'children'} />
      </MainLayout>
    );
    expect(wrapper.find('.children').length).toEqual(1);
  });
});
