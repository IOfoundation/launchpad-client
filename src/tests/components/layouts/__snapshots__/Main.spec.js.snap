// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders snapshot of Main 1`] = `
<div>
  <Connect(Header)
    homePage={false}
  />
  <MobileFooter />
</div>
`;
